src/polyfill/RTCDataChannel.ts

```diff
@@ -171,26 +179,29 @@ export default class RTCDataChannel extends EventTarget implements globalThis.RT
 
         // Needs network error, type error implemented
         if (typeof data === 'string') {
+            if (data.length > this.#dataChannel.maxMessageSize()) throw new TypeError('Max message size exceeded.')
             this.#dataChannel.sendMessage(data);
-        } else if (data instanceof Blob) {
+        } else if ('arrayBuffer' in data) {
```

Comment:
we're not actually interested in blob, but its ab method, this is important for Blob-like libraries